scale_build/bootstrap/hash.py

```diff
@@ -0,0 +1,40 @@
+import hashlib
+import logging
+import os
+import requests
+import urllib.parse
+
+from scale_build.exceptions import CallError
+
+from .utils import CHROOT_BASEDIR, get_apt_preferences, get_manifest, run
+
+
+to_disable = ('requests', 'urllib3')
+for name in filter(lambda k: k.startswith(to_disable), logging.Logger.manager.loggerDict.keys()):
+    custom_logger = logging.getLogger(name)
+    custom_logger.disabled = True
```

Comment:
Why disable a logger instead of just setting an appropriate log level? Why are you walking all loggers instead of just using propagation?

Reply:
I have tried disabling propagation but it did not help. And our default log level is debug, so these logged messaged are still caught by us.

Reply:
`logging.getLogger("requests").setLevel(logging.INFO)` should do the job

Reply:
It does not
```
[2021-04-23 04:00:35] Starting new HTTP connection (1): apt.tn.ixsystems.com:80
[2021-04-23 04:00:35] http://apt.tn.ixsystems.com:80 "GET /apt-direct/unstable/debian/dists/bullseye/Release HTTP/1.1" 200 7522
[2021-04-23 04:00:35] Starting new HTTP connection (1): apt.tn.ixsystems.com:80
[2021-04-23 04:00:36] http://apt.tn.ixsystems.com:80 "GET /apt-direct/unstable/debian-sid/dists/sid/Release HTTP/1.1" 200 7511
[2021-04-23 04:00:36] Starting new HTTP connection (1): apt.tn.ixsystems.com:80
[2021-04-23 04:00:36] http://apt.tn.ixsystems.com:80 "GET /apt-direct/unstable/docker/dists/buster/Release HTTP/1.1" 200 3486
[2021-04-23 04:00:36] Starting new HTTP connection (1): apt.tn.ixsystems.com:80
[2021-04-23 04:00:37] http://apt.tn.ixsystems.com:80 "GET /apt-direct/unstable/gluster/dists/bullseye/Release HTTP/1.1" 200 3446
[2021-04-23 04:00:37] Starting new HTTP connection (1): apt.tn.ixsystems.com:80
[2021-04-23 04:00:37] http://apt.tn.ixsystems.com:80 "GET /apt-direct/unstable/libnvidia/dists/bullseye/Release HTTP/1.1" 200 3483
[2021-04-23 04:00:37] Starting new HTTP connection (1): apt.tn.ixsystems.com:80
[2021-04-23 04:00:38] http://apt.tn.ixsystems.com:80 "GET /apt-direct/unstable/nvidia-container/dists/bullseye/Release HTTP/1.1" 200 3495
[2021-04-23 04:00:38] Starting new HTTP connection (1): apt.tn.ixsystems.com:80
[2021-04-23 04:00:38] http://apt.tn.ixsystems.com:80 "GET /apt-direct/unstable/nvidia-docker/dists/bullseye/Release HTTP/1.1" 200 3481
[2021-04-23 04:00:38] Starting new HTTP connection (1): apt.tn.ixsystems.com:80
```

Reply:
It does
```
import logging
import requests
logging.basicConfig(level=logging.DEBUG)
logging.getLogger("urllib3").setLevel(logging.INFO)
requests.get("http://google.com")
```
Prints nothing